query-engine/connectors/mongodb-query-connector/src/query_strings.rs

```diff
@@ -0,0 +1,84 @@
+//! query_strings provides functions for building query strings aynchronously from values
+//! of driver types such as Document. These query strings are not feed trough the wire to
+//! mongodb their main purpose is to add the query to log traces
+use mongodb::bson::{Bson, Document};
+use std::fmt::Write;
+
+macro_rules! write_indented {
+    ($buffer:expr, $depth:expr, $fmt_str:literal, $($args:expr)*) => {
+        write!($buffer, "{}{}", indent($depth), format!($fmt_str, $($args)*))?;
+    };
+}
+
+pub(crate) async fn build_aggregate(stages: Vec<Document>, coll_name: &str) -> String {
+    let mut buffer = String::new();
+
+    write!(buffer, "db.{}.aggregate(", coll_name).unwrap();
+
+    let stages: Vec<_> = stages.iter().map(|stage| Bson::Document(stage.clone())).collect();
+
+    fmt_list(&mut buffer, &stages, 1).unwrap();
+    write!(buffer, ")").unwrap();
+
+    return buffer.to_string();
```

Comment:
😂 took me ages to get use to not having to return at the end of a rust function. Now whenever I write javascript I forget to add the return and cannot figure out why its not working as expected.